enhancements/cluster-logging/cluster-logging-v2-apis.md

```diff
@@ -0,0 +1,429 @@
+* Support an API to spec a Red Hat managed LokiStack with the logging tenancy model
+* Support an API to allow flexible deployment of the logging components: collector/forwarder, storage, visualization
+* Continue to allow deployment of a log forwarder to the output sinks of the administrators choosing
+
+
+### Non-Goals
+
+* "One click" deployment of a full logging stack as provided by **ClusterLogging** v1
+* Complete backwards compatibility to **ClusterLogForwarder** v1
+* Automated migration path from v1 to v2
+
+## Proposal
+
+This is where we get down to the nitty gritty of what the proposal
+actually is. Describe clearly what will be changed, including all of
+the components that need to be modified and how they will be
+different. Include the reason for each choice in the design and
+implementation that is proposed here, and expand on reasons for not
+choosing alternatives in the Alternatives section at the end of the
+document.
+
+### Workflow Description
+
+The following workflow describes the first user story which is a superset of the others and allows deployment of a full logging stack to collect and forward logs to a Red Hat managed log store.
+
+**cluster administrator** is a human responsible for:
+
+* Managing and deploying day 2 operators
+* Managing and deploying an on-cluster LokiStack
+* Managing and deploying a cluster-wide log forwarder
+
+**obervability-operator** is an operator responsible for:
+
+* managing and deploying observability plugins (e.g log-view-plugin)
+
+**loki-operator** is an operator responsible for managing a loki stack
+
+**cluster-logging-operator** is an operator responsible for managing log collection and forwarding
+
+The cluster administrator does the following:
+
+1. Deploys the Red Hat **observability-operator**
+1. Deploys the Red Hat **loki-operator**
+1. Deploys an instance of **LokiStack** in the `openshift-logging` namespace
+1. Deploys the Red Hat **cluster-logging-operator**
+1. Creates a **ClusterLogForwarder** custom resource for the **LokiStack**
+
+The **loki-operator**:
+
+1. Deploys the logging-view-plugin for reading logs in the OpenShift console
+
+The **cluster-logging-operator**:
+
+1. Deploys the log collector to forward logs to log storage in the `openshift-logging` namespace
+
+### API Extensions
+
+CustomResourceDefinition:
+```yaml
+    apiVersion: "logging.openshift.io/v2"
+    kind: ClusterLogForwarder
+    metadata:
+      name: 
+    spec:
+      serviceAccount:
+        name: 
+        namespace:       #namespace of deployment and resources
+      inputs:
+      - name:
+        type:
+        application:
+          selector:  #labelselector
+          namespaces:
+            include: []
+            exclude: []
+          containers:
+            include: []
+            exclude: []
+            maxRecordsPerSecond: # map[containername] int 
+        infrastructure:
+          sources: []
+        audit:
+          sources: []
```

Comment:
Need a bit more explanation on how sources and application logs work in the new API (app logs can include what we call infra, node vs. container sources) but the API is correct.
Should clarify that the pre-defined input names "application", "infrastructure" and "audit" still have the same meaning as now, but the user can define new kinds of app/infra/audit categories of their own by defining new inputs using the API above.

Reply:
+1, I meant to say define new inputs for new "logical categories" not to define new category keywords.